### PR TITLE
STOR-1065: Rework sidecar bindings to bind common ClusterRoles

### PR DESCRIPTION
**Pull request by mpatlasov**
PR https://github.com/openshift/cluster-storage-operator/pull/379 publishes builiding blocks of sidecar ClusterRoles. Now, aws-ebs csi driver operator may compose its sidecars ClusterRoles from those building blocks.

This PR also moves permissions for `leases` resource from ClusterRole to per-namespace Role (`assets/rbac/lease_leader_election_role.yaml`).

### CODE REVIEW

**Comment by mpatlasov** [comment]
Many CI tests are likely to fail because this PR depends on https://github.com/openshift/cluster-storage-operator/pull/379

**Comment by mpatlasov** [comment]
New force-push updates PR:
 * reworked sidecar ClusterRoleBindings to match new version of https://github.com/openshift/cluster-storage-operator/pull/379 (see https://github.com/openshift/cluster-storage-operator/pull/379#issuecomment-1601898620)

**Comment by mpatlasov** [comment]
This PR passes e2e tests in conjunction with [STOR-1065: Publish ClusterRoles for csi driver sidecars](https://github.com/openshift/cluster-storage-operator/pull/379):
```
job <https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/release-openshift-origin-installer-launch-aws-modern/1671702027480076288 | test e2e https://github.com/openshift/cluster-storage-operator/pull/379,https://github.com/openshift/aws-ebs-csi-driver-operator/pull/244 aws> succeeded
```
(https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/release-openshift-origin-installer-launch-aws-modern/1671702027480076288)

File: assets/rbac/lease_leader_election_role.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Role for accessing metrics exposed by the operator
```
[the remark this answers is not in the document]
Comment (mpatlasov) [reply assets/rbac/lease_leader_election_role.yaml]: fixed!

**Comment by mpatlasov** [comment]
New force-push updates PR:
 * Fixed comments in `assets/rbac/lease_leader_election_role.yaml` and `assets/rbac/lease_leader_election_rolebinding.yaml`.

**Comment by ropatil010** [comment]
/label qe-approved
Test results updated here: [STOR-1363](https://issues.redhat.com/browse/STOR-1363#:~:text=Red%20Hat%20Employee-,Platform%3A%20AWS%20EBS,-%2B%20EFS(Hypershift%20Mgmt)

**Pull request merged**